provider/core/src/resource.rs

```diff
@@ -383,6 +383,18 @@ impl fmt::Display for ResourceOptions {
     }
 }
 
+impl Ord for ResourceOptions {
+    fn cmp(&self, other: &Self) -> Ordering {
+        self.strict_cmp(other.to_string().as_bytes())
```

Comment:
Issue: we should not call `to_string()`, which is an expensive function, in `cmp`, which should be efficient since it may be called many times.

Where do we need `impl Ord for ResourceOptions`? Maybe we can change the call sites instead.

Reply:
Maybe we can switch to HashMap as you commented below. This is where I am getting errors for ResourceOptions if Ord impl is not provided.

https://github.com/unicode-org/icu4x/blob/main/provider/core/src/hello_world.rs#L110

https://github.com/unicode-org/icu4x/blob/main/provider/core/src/hello_world.rs#L127

Probably because HelloWorldProvider uses LiteMap: https://github.com/unicode-org/icu4x/blob/main/provider/core/src/hello_world.rs#L127

Reply:
Oh, it can't, for reasons Rob mentioned

We could instead just use a manual LiteMap, a `Vec` of tuples that we binary search

Reply:
I moved all of datagen to LiteMap a while ago in part because LiteMap has a bunch of utility functions that are useful for us that don't exist in HashMap.

This is a place where LocaleStr would be a good type to have.

I'm not quite sure what to recommend to unblock @snktd.

Reply:
I don't think any of those utility functions you mentioned are still in use. See #2152, which removes `LiteMap` deps and unblocks this.